cmd/build/command.go

```diff
@@ -80,21 +89,23 @@ func Build(ctx context.Context) *cobra.Command {
 	return cmd
 }
 
-func (bc *Command) getManifestAndBuildVersion(options *types.BuildOptions) (*model.Manifest, bool) {
-	manifest, errManifest := bc.GetManifest(options.File)
-	if errManifest != nil {
-		oktetoLog.Debug("error getting manifest v2 from file %s: %v. Fallback to build v1", options.File, errManifest)
+func (bc *Command) getManifest(options *types.BuildOptions) (*model.Manifest, error) {
+	manifest, err := bc.GetManifest(options.File)
+	if err != nil {
+		if errors.Is(err, oktetoErrors.ErrInvalidManifest) {
+			return nil, err
+		}
+
+		oktetoLog.Warning("error getting manifest v2 from file %s: %v. Fallback to build v1", options.File, err)
 	}
 
-	isBuildV2 := errManifest == nil &&
-		manifest.IsV2 &&
-		len(manifest.Build) != 0
-	return manifest, isBuildV2
+	return manifest, nil
 }
 
-func (bc *Command) getBuilder(isBuildV2 bool) Builder {
-	if isBuildV2 {
-		return buildv2.NewBuilder(bc.Builder, bc.Registry)
+func isBuildV2(m *model.Manifest) bool {
+	if m == nil {
```

Comment:
this has to do with whether to allow fallback or not. As I said I'm not sure what use cases we currently cover with it so for now that logic is allowed. Basically what we are considering as an error to not continue with the execution of the command is that the manifest oktet given by the user is invalid, then we believe we are in a situation not expected by the user and we return the error. On the other hand, the manifest may be null because it does not exist, but this situation does not generate an error for the user. That is why the logic that checks that the manifest == nil after the error check, because at the moment a null manifest because the user has not created it does not cause an error that terminates the process.

Reply:
Yes, I feel like nil is the wrong way to control that logic. I'd expect something like a `errors.Is(err, v2NotDectected)`